source/controllers/src/impedance/Impedance.cpp

```diff
@@ -26,6 +26,7 @@ template<class S>
 S Impedance<S>::compute_command(const S&, const S&) {
   throw exceptions::NotImplementedException(
       "compute_command(desired_state, feedback_state) not implemented for this input class");
+  return S();
```

Comment:
This is just to suppress a compiler warning I imagine?

Reply:
yes indeed. I have it sometimes and sometimes not so I figured I should just add it as it does not harm. Well except if S does not have an empty constructor then it can become problematic

Reply:
I tried without and the compiler does not issue a warning anymore. I am certain I had one but I guess it was something else. Thus I have removed them. If we want to refactor `state_representation` to avoid mutable design we will need to remove empty constructors and it would create problem otherwise.